executor/builder.go

```diff
@@ -1528,6 +1531,14 @@ func (b *executorBuilder) getSnapshotTS() (uint64, error) {
 	return txnManager.GetStmtReadTS()
 }
 
+func (b *executorBuilder) getSnapshot() (kv.Snapshot, error) {
```

Comment:
I think it's better to save snapshot in executorBuilder in `newDataReaderBuilder` like `snapshotTS` . Because `txnManager.GetForUpdateSnapshot/GetReadSnapshot` is not thread safe too.